src/intercept-logs.ts

```diff
@@ -0,0 +1,114 @@
+ * place through the LSP connection (once it is set up).
+ *
+ * As it is common for modules to instantiate their loggers upon import,
+ * this log interception must take place as a side effect of importing this
+ * module, and this module should be imported before any others.
+ */
+
+import * as plylog from 'plylog';
+import * as util from 'util';
+import {RemoteConsole} from 'vscode-languageserver';
+import * as winston from 'winston';
+
+/**
+ * Forwards logs through the LSP connection once it is set up.
+ *
+ * This is written somewhat strangely as winston seems to require that it be an
+ * ES5-style class with inheritance through util.inherits.
+ */
+interface ForwardingTransport extends winston.TransportInstance {}
+interface ForwardingTransportStatic {
+  new (options: any): ForwardingTransport;
+
+  console: RemoteConsole|undefined;
+}
+const ForwardingTransport = function(this: ForwardingTransport, _options: any) {
+} as any as ForwardingTransportStatic;
+util.inherits(ForwardingTransport, winston.Transport);
+ForwardingTransport.console = undefined;
+
+ForwardingTransport.prototype.log = function(
+    this: ForwardingTransport, level: plylog.Level, msg: string, _meta: any,
+    callback: (err: Error|null, success: boolean) => void) {
+  if (typeof msg !== 'string') {
+    msg = util.inspect(msg);
+  }
+  const console = ForwardingTransport.console;
+  if (!console) {
+    // TODO(rictic): store these calls in memory and send them over when the
+    //     console link is established.
+    return;
+  }
+  switch (level) {
+    case 'debug':
+    case 'silly':
+    case 'info':
+    case 'verbose':
+      console.info(msg);
+      break;
+    case 'warn':
+    case 'error':
+      console.warn(msg);
+      break;
+    default:
+      const never: never = level;
+      console.warn(`Got an unknown log level: ${never}`);
+      console.warn(msg);
+  }
+  callback(null, true);
+};
+
+plylog.defaultConfig.transportFactory = (options) =>
+    new ForwardingTransport(options);
+
+export function hookUpRemoteConsole(console: RemoteConsole) {
+  if (ForwardingTransport.console) {
+    throw new Error('set remote console twice!');
+  }
+  ForwardingTransport.console = console;
+}
+
+/**
+ * A useful function for debugging. Logs through the connect to the editor
+ * so that the logs are visible.
+ */
+export function log(...args: any[]): void;
+export function log() {
+  if (!ForwardingTransport.console) {
+    return;
+  }
+  const args: any[] = Array.prototype.slice.call(arguments);
+  const strArgs = args.map((arg) => {
+    if (typeof arg === 'string') {
+      return arg;
+    }
+    return util.inspect(arg);
+  });
+  ForwardingTransport.console.log(strArgs.join(' '));
+}
+
+/** Replace console.log, warn, debug, etc with our safe version. */
+console.log = log;
+console.warn = log;
+console.error = log;
+console.debug = log;
+console.info = log;
+console.dir = log;
```

Comment:
So silly is a `winston` log level (well, it's an `npm` log level `plylog` supports because `winston` supports it). But there's no `console.silly`, it just has the above list.